File: app/src/main/java/com/example/sudoajay/duplication_data/SdCard/SdCardDialog.java
package com.example.sudoajay.duplication_data.SdCard;

import android.annotation.SuppressLint;
import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;

import com.example.sudoajay.duplication_data.Permission.AndroidSdCardPermission;
import com.example.sudoajay.duplication_data.R;


/**
 * Created by sudoajay on 4/15/18.
 */

public class SdCardDialog extends DialogFragment {

    private AndroidSdCardPermission android_sdCard_permission;

    public SdCardDialog() {

    }

    @SuppressLint("ValidFragment")
    public SdCardDialog(AndroidSdCardPermission android_sdCard_permission) {
        this.android_sdCard_permission = android_sdCard_permission;
    }

    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        try {
            View rootview = inflater.inflate(R.layout.activity_custom_dialog_sd_select, container, false);

            Button continue_Button = rootview.findViewById(R.id.continue_Button);
            Button see_More_button = rootview.findViewById(R.id.see_More_button);

            continue_Button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {

                    android_sdCard_permission.Storage_Access_FrameWork();

                    dismiss();

                }
            });

            see_More_button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    try {
                        final String url = "https://developer.android.com/guide/topics/providers/document-provider.html";
                        Intent i = new Intent(Intent.ACTION_VIEW);
                        i.setData(Uri.parse(url));
                        startActivity(i);
                    } catch (Exception ignored) {

                    }
                }
            });

            setCancelable(false);
            return rootview;
        } catch (Exception e) {
            return null;

        }


    }

    public void onStart() {
        // This MUST be called first! Otherwise the view tweaking will not be present in the displayed Dialog (most likely overriden)
        super.onStart();

    }

    @Override
    public void onDismiss(DialogInterface dialog) {
        super.onDismiss(dialog);
    }
}

File: app/src/main/java/com/sudoajay/duplication_data/sdCard/SdCardDialog.kt
package com.sudoajay.duplication_data.sdCard

import android.annotation.SuppressLint
import android.content.DialogInterface
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.DialogFragment
import com.sudoajay.duplication_data.R
import com.sudoajay.duplication_data.permission.AndroidSdCardPermission

/**
 * Created by sudoajay on 4/15/18.
 */
class SdCardDialog : DialogFragment {
    private var android_sdCard_permission: AndroidSdCardPermission? = null

    constructor() {}
    @SuppressLint("ValidFragment")
    constructor(android_sdCard_permission: AndroidSdCardPermission?) {
        this.android_sdCard_permission = android_sdCard_permission
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return try {
            val rootview = inflater.inflate(R.layout.activity_custom_dialog_sd_select, container, false)
            val continueButton = rootview.findViewById<Button>(R.id.continue_Button)
            val seeMoreButton = rootview.findViewById<Button>(R.id.see_More_button)
            continueButton.setOnClickListener {
                android_sdCard_permission!!.storageAccessFrameWork()
                dismiss()
            }
            seeMoreButton.setOnClickListener {
                try {
                    val url = "https://developer.android.com/guide/topics/providers/document-provider.html"
                    val i = Intent(Intent.ACTION_VIEW)
                    i.data = Uri.parse(url)
                    startActivity(i)
                } catch (ignored: Exception) {
                }
            }
            isCancelable = false
            rootview
        } catch (e: Exception) {
            null
        }
    }

    override fun onStart() { // This MUST be called first! Otherwise the view tweaking will not be present in the displayed Dialog (most likely overriden)
        super.onStart()
    }

    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)
    }
}